### ISSUE TITLE
[Usage]: long prompt logits calculation

### ISSUE THREAD

**Issue by Liu-yuliang** [vllm-project/vllm#4005]
### Your current environment

vllm 0.4.0
GPU=A100

### How would you like to use vllm

I need to calculate very long prompt logits like 40k in my usage, I have set gpu_memory_utilization as minimum to support 40k, but softmax and other calculations are still beyond the capacity of a single GPU, how can I adjust it.

I have deleted a lot of unneeded calculations that take up the GPU and still have the same problem with tensor_parallel_size=2

Thank you!
